ls1intum/Artemis
# `Assessment`: Improved assesment view for instructors

### Checklist
#### General


#### Client
guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).

### Motivation and Context
This PR is linked to #4176.
After assessing a single submission, the button for importing this submission as an example submission appears directly under the cursor for submitting the assessment and cause instructors to accidently click on it.

### Description
The button position for importing as example submission has been moved. In addition a dialog window has been added that opens after clicking the button in order to confirm the importing process.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Student
- Text Exercise with all dates set, so that one has enough time to start and submit the exercise and asses

1. Log in to Artemis
2. Navigate to course overview
3. Start the created text exercises
4. Submit any text
5. Wait for the submission period to be over
6. Navigate to course management, and the corresponding text exercise
7. Open submission dashboard
8. Click the button "Assess submission"
9. Add Feedback and submit
10. The buttons should be aligned correctly. When clicking the import button, the instructor should be prompted with a verification dialog. If aborted, the submission should not be imported. If approved, the submission should be imported.

### Review Progress
#### Code Review
#### Manual Tests

### Test Coverage
assessment-header.component.ts: 92.3% | 78.26%

### Screenshots
Before:
<img width="759" alt="Bildschirmfoto 2021-11-30 um 00 05 19" src="https://user-images.githubusercontent.com/73833780/143956621-0134f2ce-43a0-4439-b4ca-dd6e8cc5159f.png">

After:
<img width="1068" alt="Bildschirmfoto 2021-11-30 um 00 04 38" src="https://user-images.githubusercontent.com/73833780/143956558-1067298f-de2f-49ec-9c17-69f9c6d81e19.png">

Note: This PR hotfixes the input value 'isAtLeastInstructor' for the component assessment-layout.component.ts. As this value is redundantly defined as an attribute in the input object exercise, the input 'isAtLeastInstructor' should be replaced with the usage of the attribute in exercise. Follow up in #4428.
